datafusion/optimizer/src/push_down_filter.rs

```diff
@@ -785,7 +782,7 @@ impl OptimizerRule for PushDownFilter {
                 let mut keep_predicates = vec![];
                 let mut push_predicates = vec![];
                 for expr in predicates {
-                    let cols = expr.to_columns()?;
+                    let cols = expr.column_refs();
```

Comment:
This is a pretty good example where there is no need to copy `Column`s simply to check if they are referenced.